modules/light-clients/14-avalanche/client_state.go

```diff
@@ -283,8 +342,335 @@ func IsMatchingClientState(subject, substitute ClientState) bool {
 	return reflect.DeepEqual(subject, substitute)
 }
 
-func (cs *ClientState) VerifyUpgradeAndUpdateState(ctx sdk.Context, cdc codec.BinaryCodec, store sdk.KVStore, newClient exported.ClientState, newConsState exported.ConsensusState, proofUpgradeClient, proofUpgradeConsState []byte) error {
-	//TODO implement me
+func (cs *ClientState) VerifyUpgradeAndUpdateState(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, upgradedClient exported.ClientState, upgradedConsState exported.ConsensusState, proofUpgradeClient, proofUpgradeConsState []byte) error {
```

Comment:
Recommend refactoring this function to rely on `VerifyMembership` similar to how the tendermint light client does.

Reply:
update

Reply:
update